### DISCUSSION TITLE
Writing driver for Andor kymera Spectrograph

### DISCUSSION THREAD

**Discussion by Sounak-QSi** [microsoft/Qcodes#5148] [Q&A]
I am actually trying to write a driver for the Andor kymera spectrograph. This is a dll based instrument. 
I found a qcodes driver of [Andor_DU401](https://qcodes.github.io/Qcodes_contrib_drivers/_modules/qcodes_contrib_drivers/drivers/Andor/DU401.html#Andor_DU401) and was trying to replicate the pattern. But didnt find the required get and set commands.
      The source code maybe in the dll file, but I could not decompile it.
Here is the link of the [python wrappers of the intrument](https://drive.google.com/drive/folders/1fB398XiUl-OeS0zE7SRo1up6HjiWKFqI?usp=share_link).

If anyone can help me in this regard I will be very grateful.

Thanks in advance.

**Comment by jenshnielsen** [comment]
Usually these type of drivers are written by looking at the documentation of the instrument which normally explains which commands are available in the dll. E.g the atmcd64d class is implemented by looking at the docs of the instrument. I would recommend spending a bit of time with a ctypes tutorial to understand how to call various functions from a dll
